File: clients/algoliasearch-client-java-2/algoliasearch-core/com/algolia/model/Anchoring.java
package com.algolia.model;

import com.google.gson.TypeAdapter;
import com.google.gson.annotations.JsonAdapter;
import com.google.gson.stream.JsonReader;
import com.google.gson.stream.JsonWriter;
import java.io.IOException;

/**
 * Whether the pattern parameter must match the beginning or the end of the query string, or both,
 * or none.
 */
@JsonAdapter(Anchoring.Adapter.class)
public enum Anchoring {
  IS("is"),

  STARTSWITH("startsWith"),

  ENDSWITH("endsWith"),

  CONTAINS("contains");

  private String value;

  Anchoring(String value) {
    this.value = value;
  }

  public String getValue() {
    return value;
  }

  @Override
  public String toString() {
    return String.valueOf(value);
  }

  public static Anchoring fromValue(String value) {
    for (Anchoring b : Anchoring.values()) {
      if (b.value.equals(value)) {
        return b;
      }
    }
    throw new IllegalArgumentException("Unexpected value '" + value + "'");
  }

  public static class Adapter extends TypeAdapter<Anchoring> {

    @Override
    public void write(final JsonWriter jsonWriter, final Anchoring enumeration)
      throws IOException {
      jsonWriter.value(enumeration.getValue());
    }

    @Override
    public Anchoring read(final JsonReader jsonReader) throws IOException {
      String value = jsonReader.nextString();
      return Anchoring.fromValue(value);
    }
  }
}

File: clients/algoliasearch-client-php/lib/Model/Search/Anchoring.php
<?php

namespace Algolia\AlgoliaSearch\Model\Search;

/**
 * Anchoring Class Doc Comment
 *
 * @category Class
 * @description Whether the pattern parameter must match the beginning or the end of the query string, or both, or none.
 *
 * @package Algolia\AlgoliaSearch
 */
class Anchoring
{
    /**
     * Possible values of this enum
     */
    const IS = 'is';

    const STARTS_WITH = 'startsWith';

    const ENDS_WITH = 'endsWith';

    const CONTAINS = 'contains';

    /**
     * Gets allowable values of the enum
     *
     * @return string[]
     */
    public static function getAllowableEnumValues()
    {
        return [
            self::IS,
            self::STARTS_WITH,
            self::ENDS_WITH,
            self::CONTAINS,
        ];
    }
}

